### PR TITLE
Final Merge

### PR DESCRIPTION
**Pull request by vjspranav**
A working simulation final merge into main

### CODE REVIEW

File: experiment/references.md [hunk 1]
```diff
@@ -1 +1,5 @@
-### Link your references in here
\ No newline at end of file
+- FLUID MECHANICS - RK BANSAL
+
+- EXPERIMENTS ON FLUID MECHANICS - SARABJIT SINGH 
+
+- [The constructor](http://theconstructor.org/)
```
Comment (raj-vlabs) [inline experiment/references.md]: Please use https links exclusively

**Review by raj-vlabs — COMMENTED** [review]
1. Rename startAn to use a more meaningful name.
2. Use only https links

**Comment by vjspranav** [comment]
> 1. Rename startAn to use a more meaningful name.
> 2. Use only https links

Have done the suggested, Please confirm and merge.
Thank You.

File: experiment/simulation/js/main.js [hunk 2]
```diff
@@ -1 +1,69 @@
...
+    path.setAttribute("offset", curPosition);
+    await sleep(0.5);
+  }
+};
+
+let animation = async () => {
+  for (let i = 0; i < ids.length; i++) {
+    id = ids[i];
+    let path = document.getElementById(id);
+    flags = [true, true];
```
Comment (raj-vlabs) [inline experiment/simulation/js/main.js]: Where's flags declaration?

File: experiment/pretest.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    },
+    {
+       "question":"2. Angle of contraction is more than angle of diversion. ",
+       "answers":{
+          "a":"True",
+          "b":"False"
+       },
+       "correctAnswer":"b"
+    }
+ ]
```
Comment (raj-vlabs) [inline experiment/pretest.json:18]: Multiple extra spaces at the end of questions and options.

File: experiment/posttest.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    },
+    {
+       "question":"3. Coefficient of discharge is calculated by C_d=Q_a/Q_t.  ",
+       "answers":{
+          "a":"True ",
+          "b":"False"
+       },
+       "correctAnswer":"a"
+    }
+ ]
```
Comment (raj-vlabs) [inline experiment/posttest.json:26]: Multiple extra spaces at the end of questions and options.

**Review by raj-vlabs — CHANGES_REQUESTED** [review]
1. Multiple extra spaces at the end of questions and options in both pretest.json and posttest.json.
2. Missing declaration for flags in main.js

File: experiment/simulation/js/main.js [hunk 5]
```diff
@@ -1 +1,69 @@
...
+  tube_ids.forEach((element) => {
+    let path = document.getElementById(element);
+    path.setAttribute("offset", 0);
+  });
+  ids.forEach((ele) => {
+    let path = document.getElementById(ele);
+    path.setAttribute("offset", 0);
+  });
+};
+disablestart = false;
```
Comment (raj-vlabs) [inline experiment/simulation/js/main.js:60]: Unused variable.

File: experiment/simulation/js/main.js [hunk 6]
```diff
@@ -1 +1,69 @@
-//Your JavaScript goes in here
+let sleep = (ms) => {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+};
+
+ids = ["horizontal-tube"];
+tube_ids = ["first-tube", "second-tube"];
```
Comment (raj-vlabs) [inline experiment/simulation/js/main.js:6]: Both variables ids and tube_ids undeclared.

**Review by raj-vlabs — CHANGES_REQUESTED** [review]
1. Extra spaces still not removed from pretest.json
2. Undeclared variables ids and tube_ids and unused variable disablestart

**Pull request merged**